### ISSUE TITLE
Actor.AddScript<ChildScript>(); doesn't raise the OnEnable event

### ISSUE THREAD

**Issue by stefnotch** [FlaxEngine/FlaxEngine#507]
**Issue description:**
I called `Actor.AddScript<ChildScript>();` to create and add a new script. However, this resulted in the `public override void OnEnable()` never getting called for the child script. Unless I'm missing something, that feels like a pretty bad bug.

**Steps to reproduce:**
1. Add those two scripts to your project
```cs
public class SpawnChildScript : Script
{
    public override void OnEnable()
    {
        Debug.LogWarning("Spawning a child script");

        var childScript = Actor.AddScript<ChildScript>();
    }

}
```
```cs
public class ChildScript : Script
{
    public bool HasBeenEnabled = false;

    public override void OnEnable()
    {
        Debug.LogWarning("=====\nEnabled!\n=====");
        HasBeenEnabled = true;
    }
}
```
2. Create an actor and attach the `SpawnChildScript` to it
3. Start the game
4. Notice how the newly spawned `ChildScript` didn't get enabled

**Minimal reproduction project:**

[MissingEvent.zip](https://github.com/FlaxEngine/FlaxEngine/files/6436067/MissingEvent.zip)

**Flax version:**
Flax 1.2

**Comment by jb-perrier** [comment]
Yea, ii'm able to reproduce the error. I have investigated it a bit, on the cpp side OnEnable is called but for some reason C# code is not executed.

**Comment by mafiesto4** [comment]
The problem is that in generated C# wrapper functions we use local static variable (`IsDuringWrapperCall`) to support calling base method which breaks this in case of calling `OnEnable` during `OnEnable` call.

https://github.com/FlaxEngine/FlaxEngine/blob/5f851d2433694df84f346b09c8cce6d50bcc1a3f/Source/Tools/Flax.Build/Bindings/BindingsGenerator.Cpp.cs#L960-L975

**Comment by mafiesto4** [comment]
When debugging the code is in `Scripting.Bindings.Gen.cpp` inside generated method `OnEnable_ManagedWrapper()`

**Comment by mafiesto4** [comment]
Fixed in 67aa41afb1e7042e90ff13515e905b09fe5ac412.
